### PR TITLE
Snap cuda fullnode

### LINKED ISSUE #372: Package solana-fullnode as a Linux snap
We'd like a way to easily distribute (and automatically update) `solana-fullnode` as widely as possible.  Currently Linux users must have the full rust/cargo toolchain installed which is a barrier to entry, and makes #307 and #364 a little tougher than ideal (there should be no need to build `solana-fullnode`  *again* when deploying it to a testnet or for CI testing.

Snaps seam to fit the bill nicely: https://snapcraft.io/
We can have nightly updates for a testnet setup, multiple release channels, and snaps can be installed explicitly/manually for CI testing.

Before proceeding further, we should confirm that CUDA is workable from within the confined environment of a snap.  https://github.com/anonymouse64/cuda-samples-snap seems to imply this is possible.

* [x] Basic snap infra
* [x] CUDA snap support
* [x] Document snap usage in README

### PR DESCRIPTION
**Pull request by mvines**
Fixes #372

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -29,6 +34,44 @@ apps:
     command: solana-mint-demo
 
 parts:
+  solana-perf:
+    plugin: dump
+    build-attributes: [no-system-libraries]
+    source: https://solana-perf.s3.amazonaws.com/master/x86_64-unknown-linux-gnu/solana-perf.tgz
+    prime:
+      - solana-perf-HEAD.txt
+  solana-cuda:
```
Comment (garious) [inline snap/snapcraft.yaml]: Should this be named `solana-perf`?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -29,6 +34,44 @@ apps:
     command: solana-mint-demo
 
 parts:
+  solana-perf:
```
Comment (garious) [inline snap/snapcraft.yaml]: Should this be named something like `solana-perf-tarball`?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -5,9 +5,14 @@ description: |
...
 grade: devel
-confinement: strict
+
+# TODO: solana-perf-fullnode does not yet run with 'strict' confinement due to the
+# CUDA dependency, so use 'devmode' confinement for now
+confinement: devmode
 
 apps:
+  cuda-fullnode:
+    command: solana-cuda-fullnode
```
Comment (garious) [inline snap/snapcraft.yaml]: How about the name `solana-fullnode-perf`?

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Pull request merged**